mpcite/models.py

```diff
@@ -0,0 +1,378 @@
+    country: str = Field(default='US')
+    sponsor_org: str = Field(default='USDOE Office of Science (SC), Basic Energy Sciences (BES) (SC-22)')
+    site_url: str = Field(...)
+    contact_name: str = Field(default='Kristin Persson')
+    contact_org: str = Field(default="LBNL")
+    contact_email: str = Field(default='kapersson@lbl.gov')
+    contact_phone: str = Field(default='+1(510)486-7218')
+    related_resource: str = Field('https://materialsproject.org/citing')
+    contributor_organizations: str = Field(default='MIT; UC Berkeley; Duke; U Louvain')
+    subject_categories_code: str = Field(default='36 MATERIALS SCIENCE')
+    keywords: str = Field(...)
+    description: str = Field(default="")
+    doi: dict = Field({}, title="DOI info", description="Mainly used during GET request")
+
+    @classmethod
+    def get_title(cls, material: MaterialModel):
+        formula = material.formula_pretty
+        return 'Materials Data on %s by Materials Project' % formula
+
+    @classmethod
+    def get_site_url(cls, mp_id):
+        return 'https://materialsproject.org/materials/%s' % mp_id
+
+    @classmethod
+    def get_keywords(cls, material):
+        # keywords = '; '.join(['crystal structure',
+        #                       material.formula_pretty,
+        #                       material.chemsys,
+        #                       '; '.join(['-'.join(['ICSD', str(iid)]) for iid in material['icsd_ids']]),
+        #                       ])
+        keywords = '; '.join(['crystal structure', material.formula_pretty, material.chemsys])
+        keywords += '; electronic bandstructure' if material.bandstructure is not None else ''
+        return keywords
+
+    @classmethod
+    def get_default_description(cls):
+        return 'Computed materials data using density ' \
+               'functional theory calculations. These calculations determine ' \
+               'the electronic structure of bulk materials by solving ' \
+               'approximations to the Schrodinger equation. For more ' \
+               'information, see https://materialsproject.org/docs/calculations'
+
+    @classmethod
+    def custom_to_dict(cls, elink_record) -> dict:
+        if elink_record.osti_id is None or elink_record.osti_id == '':
+            return elink_record.dict(exclude={"osti_id", "doi"})
+        else:
+            return elink_record.dict(exclude={"doi"})
+
+
+class ElinkResponseStatusEnum(str, Enum):
+    SUCCESS = "SUCCESS"
+    FAILED = "FAILURE"
+
+
+class ELinkPostResponseModel(BaseModel):
+    osti_id: str
+    accession_num: str
+    product_nos: str
+    title: str
+    contract_nos: str
+    other_identifying_nos: Union[str, None]
+    doi: Dict[str, str]
+    status: ElinkResponseStatusEnum
+    status_message: Union[str, None]
+
+
+class DOIRecordModel(BaseModel):
+    material_id: str = Field(...)
+    doi: str = Field(default='')
+    bibtex: Union[str, None] = Field(...)
+    status: str = Field(...)
```

Comment:
I attempted this: 
```
class DOIRecordStatusEnum(Enum):
    COMPLETED = "COMPLETED"
    PENDING = "PENDING"
```
```
class DOIRecordModel(BaseModel):
   status: DOIRecordStatusEnum
```
However, when I tried to load it back from mongodb, it is going cause a lot of headache. I think the problem is that PyDantic store Enum by default as `DOIRecordStatusEnum.PENDING`, and so it would save literarily `DOIRecordStatusEnum` as a string. 

And when you load it back, it is going to look lik `DOIRecordStatusEnum.PENDING`, which will cause `DOIRecordStatusEnum.PENDING` is not in ['COMPLETED', 'PENDING'] error. 

I can just save things as a string and it saves the conversion headache, but do

Do you know of a way around it?